boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2022 Seco Spa
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/f3/stm32f303Xc.dtsi>
+#include <st/f3/stm32f303v(b-c)tx-pinctrl.dtsi>
+
+/ {
+	model = "SECO JUNO SBC-D23 board (STM32F302VCT6)";
+	compatible = "seco,stm32f3-d23";
+
+	chosen {
+		zephyr,console = &usart1;
+		zephyr,code-partition = &slot0_partition;
+		zephyr,shell-uart = &usart1;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,canbus = &can1;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led_1: led_1 {
+		    gpios = <&gpiod 8 GPIO_ACTIVE_HIGH>;
+		    label = "LED-1";
+		};
+		led_2: led_2 {
+		    gpios = <&gpiod 9 GPIO_ACTIVE_HIGH>;
+		    label = "LED-2";
+		};
+	};
+
+	out_3p3v_pwr: 3p3v-out-pwr-ctrl {
+		compatible = "regulator-fixed";
+		regulator-name = "3p3v-out-pwr-ctrl";
+		enable-gpios = <&gpioe 7 GPIO_ACTIVE_HIGH>;
+		regulator-always-on;
+		regulator-boot-on;
+		status = "okay";
+	};
+
+	out_gpio_bufa_pwr: out-gpio-bufa-pwr-ctrl {
+		compatible = "regulator-fixed";
+		regulator-name = "out-gpio-bufa-pwr-ctrl";
+		enable-gpios = <&gpiof 2 GPIO_ACTIVE_LOW>;
+		regulator-boot-on;
+		status = "okay";
+	};
+
+	out_gpio_bufb_pwr: out-gpio-bufb-pwr-ctrl {
+		compatible = "regulator-fixed";
+		regulator-name = "out-gpio-bufb-pwr-ctrl";
+		enable-gpios = <&gpiof 4 GPIO_ACTIVE_LOW>;
+		regulator-boot-on;
+		status = "okay";
+	};
+
+	in_gpio_buf_pwr: in-gpio-buf-pwr-ctrl {
+		compatible = "regulator-fixed";
+		regulator-name = "in-gpio-buf-pwr-ctrl";
+		enable-gpios = <&gpiof 6 GPIO_ACTIVE_LOW>;
+		regulator-boot-on;
+		status = "okay";
+	};
+	transceiver0: can-phy0 {
+	        compatible = "nxp,tja1040", "can-transceiver-gpio";
+	        standby-gpios = <&gpiod 0 GPIO_ACTIVE_HIGH>;
+	        max-bitrate = <1000000>;
+		#phy-cells = <0>;
+        };
+
+	aliases {
+		spi-px30 = &spi1;
+		led0 = &led_1;
+		led1 = &led_2;
+	};
+
+};
+
+&clk_hse {
+	clock-frequency = <DT_FREQ_M(8)>; /* STLink 8MHz clock */
```

Comment:
does the board come with an embedded stlink?

Reply:
No, clock is coming from an external oscillator. Removing.